lib/cached_counts.rb

```diff
@@ -135,14 +135,6 @@ def define_scope_count_attribute(attribute_name, relation, options)
       version = options.fetch :version, 1
       key = scope_count_key(attribute_name, version)
 
-      unless options.has_key?(:race_condition_fallback)
-        options[:race_condition_fallback] = default_race_condition_fallback_proc(
-          key,
-          relation,
-          options
-        )
-      end
```

Comment:
It seems that the purpose of the `race_condition_fallback` is to prevent the case where the count value is not in the cache and multiple processes attempt to read the value from the cache at the same time. Upon discovering that the value is not in the cache, all processes would then attempt to concurrently compute the count value by executing the potentially expensive query against the source database before writing the value to the cache.

The `race_condition_fallback` option attempted to avoid this scenario by having the first reader that discovers the value is not in the cache, get a "fallback" value stored in the `race_condition_fallback` proc and write that fallback into the cache. Concurrent readers attempting to access the count value at this point would find the fallback value in the cache and would therefore not attempt to execute the expensive query to compute the count. The initial reader would continue on executing the expensive query to compute the real count value and then write the updated count value to the cache once the query completed.

However, this scheme introduced several problems

* When this module is attaching new methods to classes that call `caches_count_where` during the application startup process as the application initializes class definitions, the `default_race_condition_fallback_proc` method would actually run the expensive query to compute an initial fallback value to use if the value was not found in the cache. This caused a problem when deploying a new application revision to multiple app servers because that meant multiple app servers would be executing the expensive count query concurrently during their startup process which could lead to excess load on the database. This is the exact scenario the `race_condition_fallback` option was meant to prevent.

* The `race_condition_fallback` option does not even guarantee that multiple process will not attempt to run the expensive query concurrently. The TTL of the fallback value and the [`race_condition_ttl` option](https://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-fetch) are only set to 30 seconds. Therefore if the underlying query takes longer than 30 seconds to execute, then multiple processes will be attempting to run the expensive query anyways

I did not remove the `race_condition_fallback` altogether to give callers the option of specifying their own race condition strategy and to ensure we do not break any existing consumers of this Gem

https://github.com/academia-edu/cached_counts/blob/1eda5616cc59d479fe6e68b82f3cb3ef9f4ede37/lib/cached_counts.rb#L227-L238